src/hpr_packet_reporter.erl

```diff
@@ -0,0 +1,331 @@
+            gateway_timestamp_ms = Timestamp,
+            oui = OUI,
+            net_id = NetID,
+            rssi = RSSI,
+            frequency_mhz = FrequencyMhz,
+            datarate = Datarate,
+            snr = SNR,
+            region = Region,
+            gateway = Gateway,
+            payload_hash = crypto:hash(sha256, Payload)
+        },
+        packet_router_packet_report_v1_pb
+    ).
+
+-spec open_tmp_file(string(), [atom()]) -> {ok, file:io_device()} | {error, atom()}.
+open_tmp_file(FilePath, Options) ->
+    case file:open(FilePath, [append] ++ Options) of
+        {error, Error} ->
+            lager:error("failed to open tmp write file: ~p", [Error]);
+        IODevice ->
+            IODevice
+    end.
+
+-spec upload_file(aws_client:aws_client(), string(), binary()) -> ok.
+upload_file(AWSClient, Path, FileName) ->
+    BucketName = application:get_env(hpr, packet_reporter_bucket, <<"default_bucket">>),
+    {ok, Content} = file:read_file(Path),
+    case
+        aws_s3:put_object(AWSClient, BucketName, FileName, #{
+            <<"Body">> => Content
+        })
+    of
+        {ok, _, _} ->
+            ok;
+        Error ->
+            lager:error(
+                "failed to upload packet report: file: ~p, error: ~p", [Path, Error]
+            )
+    end.
+
+-spec get_packets_by_timestamp(integer()) -> [term()].
+get_packets_by_timestamp(Timestamp) ->
+    ets:select(?ETS, [{{'$1', '$2'}, [{'<', '$1', Timestamp}], ['$2']}]).
+
+-spec delete_packets_by_timestamp(integer()) -> integer().
+delete_packets_by_timestamp(Timestamp) ->
+    ets:select_delete(?ETS, [{{'$1', '$2'}, [{'<', '$1', Timestamp}], [true]}]).
+
+% ------------------------------------------------------------------
+% EUNIT Tests
+% ------------------------------------------------------------------
+-ifdef(TEST).
+-include_lib("eunit/include/eunit.hrl").
+-include_lib("common_test/include/ct.hrl").
+
+file_test() ->
+    ?ETS = ets:new(?ETS, [named_table, bag, public, {write_concurrency, true}]),
+    Timestamp = erlang:system_time(millisecond),
+    FilePath = "./data/tmp/packetreport." ++ integer_to_list(Timestamp) ++ ".txt",
+
+    Packet = test_utils:join_packet_up(#{}),
+    Packet2 = test_utils:uplink_packet_up(#{}),
+    Route = test_utils:packet_route(#{}),
+
+    report_packet(Packet, Route),
+    report_packet(Packet2, Route),
+
+    timer:sleep(1000),
+
+    handle_write(FilePath, []),
+
+    {ok, Content} = file:read_file(FilePath),
+
+    2 = length(binary:split(Content, <<"\n">>, [trim, global])),
+
+    Packet3 = test_utils:join_packet_up(#{}),
+    Packet4 = test_utils:uplink_packet_up(#{}),
+
+    report_packet(Packet3, Route),
+    report_packet(Packet4, Route),
+
+    timer:sleep(1000),
+
+    handle_write(FilePath, []),
+
+    {ok, Content2} = file:read_file(FilePath),
+
+    4 = length(binary:split(Content2, <<"\n">>, [trim, global])),
+
+    file:delete(FilePath),
+
+    ok.
+
+% aws_test() ->
```

Comment:
If this is for testing with the localstack setup, can we some documentation around that?
If it's not, and unused, this can be removed.